src/plugins/console/public/application/components/top_nav_menu.tsx

```diff
@@ -29,22 +29,77 @@
  */
 
 import React, { FunctionComponent } from 'react';
-import { EuiTabs, EuiTab } from '@elastic/eui';
+import { EuiTabs, EuiTab, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+
+interface CommonProps {
+  disabled: boolean;
+  onClick: () => void;
+  ['data-test-subj']: string;
+}
+
+export enum MenuItemPosition {
+  LEFT = 'left',
+  RIGHT = 'right',
+}
 
 export interface TopNavMenuItem {
   id: string;
   label: string;
   description: string;
   onClick: () => void;
   testId: string;
+  render?: (commonProps: CommonProps) => React.JSX.Element;
+  position: MenuItemPosition;
```

Comment:
maybe add order for the item?

Reply:
May not because we are adding these items in an array and all the items come from a single file, not a registration placeholder for other plugins to insert.